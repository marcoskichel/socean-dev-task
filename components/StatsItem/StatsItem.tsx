import { LinearProgress, styled, Typography } from '@mui/material';
import { LiveStats } from '../../providers/LiveStatsProvider';

const Root = styled('div')(({ theme }) => ({
  padding: '2rem',
  border: '1px solid #171717',
  borderRadius: '1.5rem',
  flexBasis: 0,
  flexGrow: 1,
  [theme.breakpoints.up('md')]: {
    borderRadius: 0,
  },
  '&:first-child': {
    borderTopLeftRadius: '1.5rem',
    borderBottomLeftRadius: '1.5rem',
  },
  '&:last-child': {
    borderTopRightRadius: '1.5rem',
    borderBottomRightRadius: '1.5rem',
  },
}));

type Props = LiveStats;

const titleSx = {
  marginBottom: '4.75rem',
};

const ProgressPlaceholder = () => <div></div>;

const StatsItem = (props: Props) => {
  const { title, value, hint, progress } = props;
  return (
    <Root>
      <Typography color="text.secondary" variant="h6" sx={titleSx}>
        {title}
      </Typography>
      {progress ? (
        <LinearProgress variant="determinate" value={progress} />
      ) : (
        <ProgressPlaceholder />
      )}
      <Typography variant="h5">{value}</Typography>
      <Typography color="text.secondary" variant="overline">
        {hint}
      </Typography>
    </Root>
  );
};

export default StatsItem;
