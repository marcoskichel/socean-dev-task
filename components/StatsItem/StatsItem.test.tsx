import { render } from '@testing-library/react';
import StatsItem from './StatsItem';

describe('StatsItem', () => {
  it('renders (required props)', () => {
    const { asFragment } = render(
      <StatsItem
        hint="Test Hint"
        title="Test Title"
        value="Test Value"
        progress={20}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders (all props)', () => {
    const { asFragment } = render(
      <StatsItem
        hint="Test Hint"
        title="Test Title"
        value="Test Value"
        progress={20}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
