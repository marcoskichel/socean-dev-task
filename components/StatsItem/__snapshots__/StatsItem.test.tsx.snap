// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsItem renders (all props) 1`] = `
<DocumentFragment>
  <div
    class="css-fi8v5m"
  >
    <h6
      class="MuiTypography-root MuiTypography-h6 css-1jgmn5n-MuiTypography-root"
    >
      Test Title
    </h6>
    <div
      class="css-1rhchyf"
    >
      <span
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="20"
        class="MuiLinearProgress-root MuiLinearProgress-colorSecondary MuiLinearProgress-determinate css-10jyip1-MuiLinearProgress-root"
        role="progressbar"
      >
        <span
          class="MuiLinearProgress-bar MuiLinearProgress-barColorSecondary MuiLinearProgress-bar1Determinate css-17282r-MuiLinearProgress-bar1"
          style="transform: translateX(-80%);"
        />
      </span>
    </div>
    <h5
      class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
    >
      Test Value
    </h5>
    <span
      class="MuiTypography-root MuiTypography-overline css-1hbyad5-MuiTypography-root"
    >
      Test Hint
    </span>
  </div>
</DocumentFragment>
`;

exports[`StatsItem renders (required props) 1`] = `
<DocumentFragment>
  <div
    class="css-fi8v5m"
  >
    <h6
      class="MuiTypography-root MuiTypography-h6 css-1jgmn5n-MuiTypography-root"
    >
      Test Title
    </h6>
    <div
      class="css-1rhchyf"
    >
      <span
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="20"
        class="MuiLinearProgress-root MuiLinearProgress-colorSecondary MuiLinearProgress-determinate css-10jyip1-MuiLinearProgress-root"
        role="progressbar"
      >
        <span
          class="MuiLinearProgress-bar MuiLinearProgress-barColorSecondary MuiLinearProgress-bar1Determinate css-17282r-MuiLinearProgress-bar1"
          style="transform: translateX(-80%);"
        />
      </span>
    </div>
    <h5
      class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
    >
      Test Value
    </h5>
    <span
      class="MuiTypography-root MuiTypography-overline css-1hbyad5-MuiTypography-root"
    >
      Test Hint
    </span>
  </div>
</DocumentFragment>
`;
